src/lualib/Error.ts

```diff
@@ -0,0 +1,93 @@
+Error = class {
+    public static captureStackTrace(targetObject: {}, depth: number): void {
+        targetObject["stack"] = debug.traceback(undefined, depth);
+    }
+
+    public message: string;
+    public name: string;
+    public stack?: string;
+
+    constructor(message?: string, name?: string) {
```

Comment:
Why `name` is a second argument there? In JS you'd usually just set `name` field in a subclass.

Reply:
As the implementation gets hidden away behind the default `Error` type, my idea was to use the name parameter to avoid redundant work and to choose and appropriate stack trace here. But if we do use `captureStackTrace` as intended it would make sense to stick with the standard definition here too.

Reply:
Actually I guess it's not that simple. In JS extended error classes don't have to use `captureStackTrace` (it has to be used only as a hack for es5 downleveling), so that's something that would be nice to avoid.

Looks like .stack in JS is a lazy getter:

```js
const error = new Error();
error.stack // Error...
error.name = 'MyError';
error.stack // Error...
```

```js
const error = new Error();
error.name = 'MyError';
error.stack // MyError...
```

Reply:
Lazy getter would be nice, however unavailable here due to #725 as it would break subclassing `Error`.